velox/exec/Driver.cpp

```diff
@@ -472,6 +472,15 @@ void Driver::run(std::shared_ptr<Driver> self) {
       "The last operator (sink) must not produce any results. "
       "Results need to be consumed by either a callback or another operator. ")
 
+  // There can be a race between Task terminating and the Driver being on the
```

Comment:
I took a look the code and wondering what's the race? The code here seems to be able to handle the race well: https://www.internalfb.com/code/fbsource/[2e5cf00f048d2a65a4068e84ce8970bfcd09c590]/fbcode/velox/exec/Task.cpp?lines=1272. Also could you add a unit test for this? Thanks!

Reply:
The same unit test is testing for the race with --stress in buck.

Reply:
After a lot of digging I believe the race is when driver is considered to be on the thread when terminate is happening, but then it exits the  thread and ends up here with the promise, while task is already terminated.
As the result close() is never called and the promise is never fulfilled and the lambda to get back on thread (and then exit right away and then destroy the driver) would never be called.
It might be that the race can also be eliminated elsewhere.

Reply:
What exact code path can lead into this? So you saying without this fix, the test you added could fail under stress test mode 100 on devserver? Thanks!